### DISCUSSION TITLE
Does imgproxy support image reconstruction on the fly?

### DISCUSSION THREAD

**Discussion by yaronuliel** [imgproxy/imgproxy#1247] [Q&A] [answered]
Image reconstruction is a measure of security that is used to filter out malicious parts in valid images

There are many known attacks that are based on having a perfectly valid image that hide inside it's raw data a malicious code (to be used in exploited system, or to transfer secret information)

when reconstructing an image in a proxy - the image is "rebuilt" into a new image that contains only the pixel data, and allowed metadata, at the proxy level

I'm new to imgproxy, and I didn't see direct mention for this feature in the docs... is it available?

**Comment by DarthSim** [comment, marked as answer]
Hey @yaronuliel!

I can't find anything on this subject on the internet, but imgproxy reencodes images unless the processing is skipped due to `skip_formats` or `raw` processing options. Also, imgproxy strips image metadata by default.
